sdk/component/component.go

```diff
@@ -0,0 +1,71 @@
+package component
+
+import (
+	"context"
+
+	ocsf "github.com/ocurity/dracon/sdk/component/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+// Helpers interfaces for common functionalities.
+type (
+	// Validator allows validating findings by a specified criteria.
+	Validator interface {
+		// Validate validates the supplied findings and returns an error if invalid.
+		Validate(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error
+	}
+
+	// Reader allows reading findings from a source.
+	Reader interface {
+		// Read reads and parses findings from a source.
+		Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error)
+	}
+
+	// Filterer allows filtering out findings by a specified criteria.
+	Filterer interface {
+		// Filter returns filtered findings from the supplied ones.
+		Filter(ctx context.Context, findings []*ocsf.VulnerabilityFinding) ([]*ocsf.VulnerabilityFinding, error)
+	}
+)
+
+// Components interfaces.
+type (
+	// Sourcer defines the behaviour of source components.
+	Sourcer interface {
+		// Source runs an arbitrary step. Useful for interacting with third-party API - i.e. cloning a repository.
+		Source(ctx context.Context) error
```

Comment:
❓ I'm not a super fan of this method name so please come up with better suggestions if you think of any

Reply:
`ProcessSource` ?
A git-sourcer clones a repo and filters paths
A wget sourcers downloads results from a path
A potential AWS sourcer sets up access to aws
A potential X sourcer ensures that X is ready for scanning